x-pack/plugins/infra/public/pages/metrics/inventory_view/components/waffle/node.tsx

```diff
@@ -46,245 +33,73 @@ interface Props {
-                  {node.name}
-                </Label>
-                <Value data-test-subj="nodeValue" color={color}>
-                  {value}
-                </Value>
-              </ValueInner>
-            ) : (
-              ellipsisMode && (
-                <ValueInner aria-label={nodeAriaLabel}>
-                  <Label color={color}>...</Label>
-                </ValueInner>
-              )
-            )}
-          </SquareInner>
-        </SquareOuter>
-      </NodeContainer>
-    );
-
-    const smallSquare = (
-      <NodeContainerSmall
-        data-test-subj="nodeContainer"
-        style={{ width: squareSize || 0, height: squareSize || 0, ...nodeBorder }}
-        onClick={this.togglePopover}
-        onMouseOver={this.showToolTip}
-        onMouseLeave={this.hideToolTip}
-        color={color}
-      />
-    );
-
-    const nodeSquare = valueMode || ellipsisMode ? bigSquare : smallSquare;
-
-    return (
-      <>
-        {isPopoverOpen ? (
-          <EuiPopover
-            button={nodeSquare}
-            isOpen={isPopoverOpen}
-            closePopover={this.closePopover}
-            anchorPosition="downCenter"
-            style={{ height: squareSize }}
-          >
-            <NodeContextMenu
-              node={node}
-              nodeType={nodeType}
-              options={options}
-              currentTime={currentTime}
-            />
-          </EuiPopover>
-        ) : isToolTipOpen ? (
-          <EuiToolTip
-            delay="regular"
-            position="right"
-            content={
-              <ConditionalToolTip currentTime={currentTime} node={node} nodeType={nodeType} />
-            }
-          >
-            {nodeSquare}
-          </EuiToolTip>
-        ) : (
-          nodeSquare
-        )}
+export const Node = ({
+  nodeType,
+  node,
+  options,
+  squareSize,
+  bounds,
+  formatter,
+  currentTime,
+}: Props) => {
+  const [isToolTipOpen, { off: hideToolTip, on: showToolTip }] = useBoolean(false);
+  const [isPopoverOpen, { off: closePopover, toggle: togglePopover }] = useBoolean(false);
+  const [{ detailsItemId }, setProperties] = useAssetDetailsFlyoutState();
+
+  const metric = first(node.metrics);
+  const rawValue = (metric && metric.value) || 0;
+  const color = colorFromValue(options.legend, rawValue, bounds);
+  const value = formatter(rawValue);
+
+  const toggleAssetPopover = () => {
+    if (nodeType === 'host') {
+      setProperties({ detailsItemId: node.name });
+    } else {
+      togglePopover();
+    }
+  };
 
-        {this.state.isOverlayOpen && (
-          <NodeContextPopover
-            openAlertFlyout={this.openAlertFlyout}
+  const nodeSquare = (
+    <NodeSquare
+      squareSize={squareSize}
+      togglePopover={toggleAssetPopover}
+      showToolTip={showToolTip}
+      hideToolTip={hideToolTip}
+      color={color}
+      nodeName={node.name}
+      value={value}
+      isOverlayOpen={!!(detailsItemId && node.name === detailsItemId)}
```

Comment:
Would this be enough?
```suggestion
      isOverlayOpen={node.name === detailsItemId}
```